### PR TITLE
Milestone 2: User interface

### PR DESCRIPTION
**Pull request by enaburekhan**
****Milestone 2: User interface on Tic-tac-toe game**
**In this project:**

- Create the game class
- Create four methods inside the game class to interface with the players

### CODE REVIEW

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone. I'm impressed with the fact that you have a customized README.md file for each milestone :clap: 
However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] In this milestone, you are expected to only **mock** the basic flow of the game. 

- This means **implementing a user interface that displays prompt for users (print/puts) and collects user input (get.chomp)**. 
- No game logic (for example checking who is a winner) is required in this milestone - instead, you can just display `The winner is X` no matter who won.

- [ ] The required basic game flow are highlighted below:
- Game tells which player turn it is (without computing who the winner is)
- Game asks the player to select from available moves
- Game informs the player if the selected move is invalid (without computing what an invalid move is)
- Game displays the board after the player move (without computing which spots on the board are taken)
- Game informs the player if the selected move is a winning move (without computing the winning move)
- Game informs the player if the selected move is a draw move (without computing draw)
- Game repeats all actions for the next player's move

**In summary, if I run your game, I should have the flow above available to me on the console.**

**[ OPTIONAL CHANGES ]**

- [ ] Your game can ask players for their names :blush: 

* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by enaburekhan** [comment]
**Code review changes effected**

- Add the necessary steps required to play the game

**Review by Rhelli — CHANGES_REQUESTED** [review]
# Changes Required 🛑♻️

Hi team,

Wow. Okay, let me give you a round of applause for your efforts so far 👏🏽.
 - Good Readme? Check. ✅
 - Clean, tidy code? Check. ✅
 - A version of Tic-Tac-Toe with a user interface? Check. ✅
 - Prep-work done for implementing the logic? Check ✅

Really guys, great work so far. I will however echo the previous TSE in stating that you should definitely stick to the Milestone objectives. I think it's awesome that you've practically made a start on some of the logic, but hitting the objectives ensures your not doing more work than you need to, and its a good practice and habit to get into.

I do have to unfortunately pick you up on two minor issues - your Rubocop disables at the top of your `main`. So...

 ### **[Mandatory]** - **Please remove the Rubocop disables at the top of your `main`. Rubocop may be annoying at times and it's rules may seem inconvenient, but they are not without reason and they are well thought out.**

image

For this 👆🏽 `MultipleComparison` error, there are quite a few ways to solve this. Using a loop pops into my head first but honestly, there are quite a few ways you could do this. Get creative! The objective here should be to make this piece of code more concise and more efficient!


******

image

As for this 👆🏽 `LiteralAsCondition` error (that most likely comes from here), just amend your loop structure to this 👇🏽 and you'll be golden. (Trust me, being able to say `game_on = true` or `game_on = false` comes in handy later.
image

******

Get these changes fixed and we'll be able to get your project approved! Once again, great work on the project so far! Good luck!

If you need any help, feel free to @ me here on Github or reach out to me on Slack.

All the best,

💂🏽‍♂️ Rory

**Comment by enaburekhan** [comment]
**Code review changes effected:**

- Remove robocop disable
- Refactor codes

**Review by ElijahTobs — APPROVED** [review]
# Status: Approved! :tada: 

## Congratulations guys!

### Good job implementing the changes requested by the previous TSE.

You can now merge and submit.

Happy Coding :computer: 

Feel free to reach out to me 
slack @ElijahTobs
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)
